client/src/components/Forms/CourseTaskSelect.tsx

```diff
@@ -1,20 +1,50 @@
 import * as React from 'react';
 import { Select, Form } from 'antd';
+import _ from 'lodash';
+
 import { CourseTask } from 'services/course';
 
-type Props = { data: CourseTask[]; onChange?: (id: number) => void };
+type Props = {
+  data: CourseTask[];
+  onChange?: (id: number) => void;
+};
+
+const MAX_DATE = '+275760-09-13T00:00:00.000Z';
 
 export function CourseTaskSelect(props: Props) {
-  const { data, onChange, ...rest } = props;
-  const selectProps = onChange ? { onChange } : {};
+  const { data, onChange, ...options } = props;
+  data.sort(function (firstTask, secondTask) {
```

Comment:
please re-assign the result into a new variable to make it obvious . e.g.

`const sortedData = data.sort(...`

Also probably you don't need to do parsing here. You can just compare strings. It's a benefit of ISO format. e.g. 
```
secondTask.studentEndDate?.localeCompare(firstTask.studentEndDate ?? MAX_DATE) ?? ...
```

Reply:
Done